lib/form1010_ezr/service.rb

```diff
@@ -153,10 +150,20 @@ def post_fill_veteran_ssn(parsed_form)
       parsed_form
     end
 
+    def post_fill_veteran_gender(parsed_form)
+      return if parsed_form['gender'].present?
+
+      StatsD.increment("#{Form1010Ezr::Service::STATSD_KEY_PREFIX}.missing_gender")
+
+      parsed_form['gender'] = @user.gender
+      parsed_form
```

Comment:
I know we do this on the other post-fill logic, but why are we returning the `parsed_form` from each of these methods if we are already mutating the passed in `parsed_form`?  I see that helps with testing the specific methods, but ultimately it makes it really hard to follow where all we are mutating our `parsed_form` before actually submitting it. Do we like that pattern? I know for me it makes it hard to follow and ultimately more error prone.

Reply:
I can understand the confusion. I refactored the code so we aren't returning `parsed_form` multiple times. Please feel free to re-review!